### DISCUSSION TITLE
How can I automatically apply a filter set by default for all profiles' repository lists?

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [community/community#147619] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

For [`147673#discussion-7724391`](https://github.com/orgs/community/discussions/147673#discussion-7724391):

> Rarely am I interested in which repositories a user has forked. Consequently, I want all profiles' repositories to display solely their "Source" repositories by default, like [`github.com/RokeJulianLockhart?tab=repositories&q=&type=source`](https://github.com/RokeJulianLockhart?tab=repositories&q=&type=source&language=&sort=).

**Comment by bmiit145** [comment, marked as answer]
### Use a Browser Extension with a Custom Script
If you'd like this behavior to be applied automatically across all profiles, you can use a browser extension like [Tampermonkey](https://www.tampermonkey.net/) to inject custom JavaScript into GitHub pages.

#### Steps:
1. Install Tampermonkey or a similar extension for your browser.
2. Create a new user script with the following code:

```javascript
// ==UserScript==
// @name         GitHub Default Source Repositories
// @namespace    http://tampermonkey.net/
// @version      1.0
// @description  Redirect to show only source repositories on GitHub profiles by default
// @author       Priyank Bharatbhai Sutariya
// @match        https://github.com/*?tab=repositories*
// @grant        none
// ==/UserScript==

(function() {
    'use strict';

    // Check if the URL already includes the "type=source" filter
    if (!window.location.search.includes("type=source")) {
        // Append or update the query parameter to include "type=source"
        const url = new URL(window.location.href);
        url.searchParams.set("type", "source");
        window.location.replace(url.toString());
    }
})();
```

3. Save and enable the script.
4. Whenever you visit a GitHub profile repository tab, the script will automatically redirect you to the filtered view showing only "Source" repositories.

**Reply by roke-julian-lockhart** [reply]
> [`#discussioncomment-11636385`](https://github.com/orgs/community/discussions/147619#discussioncomment-11636385)

Thanks, @bmiit145. I didn't realize that I accidentally filed this discussion as a question (rather than "product feedback") so I've refiled the FR at [`147673#discussion-7724391`](https://github.com/orgs/community/discussions/147673#discussion-7724391), rephrased this discussion as a question, and marked your response as its answer.
